Add CLI tutorial (#23)
svix/svix-docs: docs/cli.mdx

---
title: Getting Started with the CLI
---

import CLITabs from '@theme/CLITabs';
import TabItem from '@theme/TabItem';

The [Svix CLI](https://github.com/svix/svix-cli) is one of the fastest ways to interact with our API.

With the Svix CLI you can:
 - Interact with our API
 - Validate Webhook payloads
 - Quickly open our docs

> In this tutorial, we will learn how to install, update, and use the Svix CLI.

Reviewer: They don't care about updating. They haven't fallen in love with the CLI!
The goal with this tutorial is to make them fall in love!

```sh
brew install svix/svix/svix
```
</TabItem>
<TabItem value="scoop">

To install the Svix CLI with [Scoop](https://scoop.sh/), run:

```sh
scoop bucket add svix https://github.com/svix/scoop-svix.git
```

```sh
scoop install svix
```

</TabItem>
<TabItem value="macos">

To install the Svix CLI on macOS without Homebrew:

1. Download the latest Darwin tar.gz file from <https://github.com/svix/svix-cli/releases/latest>

2. Extract the file: `tar -xvf svix_X.X.X_Darwin_x86_64.tar.gz`

2. Optionally, install the binary in a location where you can execute it globally (e.g., /usr/local/bin).

</TabItem>
<TabItem value="linux">

To install the Svix CLI on Linux without a package manager:

1. Download the latest Linux tar.gz file from <https://github.com/svix/svix-cli/releases/latest>

2. Extract the file: `tar -xvf svix_X.X.X_Linux_x86_64.tar.gz`

3. Run the executable: `./svix`

</TabItem>
<TabItem value="windows">

To install the Svix CLI on Windows without Scoop:

1. Download the latest Windows tar.gz file from <https://github.com/svix/svix-cli/releases/latest>

2. Extract the `svix_X.X.X_Windows_x86_64.tar.gz` file.

3. Run the `.exe` file

</TabItem>
</CLITabs>

> ### 2) Login to your Svix Account

Reviewer: I forgot to mention, but this whole section is confusing. This person doesn't care about `login` yet, they are just learning on the CLI for the first time. It should be easy and straightforward as possible. I'd just mention the env var and not talk about login.

Author: I rephrased to `Pair with your Svix Account` and simplified it, do you think it makes more sense now or would you still prefer to just say "set the env variable"

To upgrade the Svix CLI with [Hombrew](https://homebrew.sh), run:

```sh
brew upgrade svix/svix/svix
```
</TabItem>
<TabItem value="scoop">

To upgrade the Svix CLI with [Scoop](https://scoop.sh/), run:

```sh
scoop update svix
```

</TabItem>
<TabItem value="macos">

To upgrade the Svix CLI on macOS without Homebrew:

1. Download the latest Darwin `tar.gz` file from <https://github.com/svix/svix-cli/releases/latest>

2. Extract the file: `tar -xvf svix_X.X.X_Darwin_x86_64.tar.gz`

2. Optionally, install the binary in a location where you can execute it globally (e.g., /usr/local/bin).

</TabItem>
<TabItem value="linux">

To upgrade the Svix CLI on Linux without a package manager:

1. Download the latest Linux `tar.gz` file from <https://github.com/svix/svix-cli/releases/latest>

2. Extract the file: `tar -xvf svix_X.X.X_Linux_x86_64.tar.gz`

3. Run the executable: `./svix`

</TabItem>
<TabItem value="windows">

To upgrade the Svix CLI on Windows without Scoop:

1. Download the latest Windows `tar.gz` file from <https://github.com/svix/svix-cli/releases/latest>

2. Extract the `svix_X.X.X_Windows_x86_64.tar.gz` file.

3. Run the `.exe` file

</TabItem>
</CLITabs>

> ### 5) Get creative

Reviewer: This is the main section of the tutorial, it needs to have more beef and merged with the section about the first command.
Don't say "run the first command", but rather something like:

First we'll create an app
`example - create app`
Let's see the app is showing
`example - list apps`
Or we can get it specifically
`example - get app`

Send message
`example - send message`

Author: Restructured to roughly follow this story.

The Svix CLI has commands to interact with every part of the Svix API.

To get a complete list of commands, run `svix --help`.

Each command accepts raw JSON input as either the first positional argument or piped in over `stdin`.

With this in mind, you can chain Svix CLI commands together with other command line tools like the popular JSON parser
[jq](https://stedolan.github.io/jq/) to create powerful snippets to improve your workflow.

For example, to automatically open the dashboard for the app we created in step 3, on macOS you can run the following command:

```sh

> svix application list |  jq '.data[] | select(.name == "My Application").id' | xargs svix auth dashboard | jq '.url' | xargs open

Reviewer: It's a bit too complex tbh, maybe change it to:
```
svix application list | jq '.data[0].id' | xargs svix auth dashboard | jq '.url' | xargs open
```

Also, there's a bug there. If the response has more than 50 apps it won't actually work.

Author: Yeah I know... It just felt more exciting to be able to do it by name instead of id. Updated it to be by ID as the tutorial requires app ids now anyways.

---
title: Getting Started with the CLI
---

import CLITabs from '@theme/CLITabs';
import TabItem from '@theme/TabItem';

> The [Svix CLI](https://github.com/svix/svix-cli) is one of the fastest way to interact with our API.

Reviewer 2: ```suggestion
The [Svix CLI](https://github.com/svix/svix-cli) is one of the fastest ways to interact with our API.
```

> With the Svix CLI you can:

Reviewer: Or better yet, just make it as a paragraph.

 - Interact with our API
 - Validate Webhook payloads
 - Quickly open our docs

> In this tutorial we will learn how to install, update, and use the Svix CLI.

Reviewer 2: ```suggestion
In this tutorial, we will learn how to install, update, and use the Svix CLI.
```

### 1) Install the Svix CLI

<CLITabs>
<TabItem value="homebrew">

To install the Svix CLI with [Hombrew](https://homebrew.sh), run:

```sh
brew install svix/svix/svix
```
</TabItem>
<TabItem value="scoop">

To install the Svix CLI with [Scoop](https://scoop.sh/), run:

```sh
scoop bucket add svix https://github.com/svix/scoop-svix.git
```

```sh
scoop install svix
```

</TabItem>
<TabItem value="macos">

To install the Svix CLI on macOS without Homebrew:

1. Download the latest Darwin tar.gz file from <https://github.com/svix/svix-cli/releases/latest>

2. Extract the file: `tar -xvf svix_X.X.X_Darwin_x86_64.tar.gz`

2. Optionally, install the binary in a location where you can execute it globally (e.g., /usr/local/bin).

</TabItem>
<TabItem value="linux">

To install the Svix CLI on Linux without a package manager:

> 

Reviewer 2: ```suggestion
```

1. Download the latest Linux tar.gz file from <https://github.com/svix/svix-cli/releases/latest>

2. Extract the file: `tar -xvf svix_X.X.X_Linux_x86_64.tar.gz`

3. Run the executable: `./svix`

</TabItem>
<TabItem value="windows">

To install the Svix CLI on Windows without Scoop:

1. Download the latest Windows tar.gz file from <https://github.com/svix/svix-cli/releases/latest>

2. Extract the `svix_X.X.X_Windows_x86_64.tar.gz` file.

3. Run the `.exe` file

</TabItem>
</CLITabs>

### 2) Login to your Svix Account

After installing the Svix CLI, you must pair it with your Svix account.

To do so, run `svix login` in the terminal. You will be prompted for your Svix Auth Token.

Once supplied, this will be stored in your configuration file.

:::info

> On most platforms, your config file can be found at `~/.config/svix/config.toml`.

Reviewer 2: Maybe?

```suggestion
On *nix systems, your config file can be found at `~/.config/svix/config.toml`.
```

On Windows, your config file can be found at `%APPDATA%/svix/config.toml`
:::

If you would prefer not to store your AUTH_TOKEN on disk you can temporarily set it via the `SVIX_AUTH_TOKEN` env variable:

 - On macOS and Linux: `export SVIX_AUTH_TOKEN=<MY-AUTH-TOKEN>`
 - On Windows via command prompt: `set SVIX_AUTH_TOKEN=<MY-AUTH-TOKEN>`
 - On Windows via PowerShell: `$env:SVIX_AUTH_TOKEN = '<MY-AUTH-TOKEN>'`

### 3) Run your first command

You’re now ready to use the Svix CLI. To get started let's create a new application:

```sh
svix application create --data-name "My Application"
```

This command creates a new application on Svix and returns the raw application object as JSON.

> ### 4) Upgrade the Svix CLI

Reviewer: I think it's probably not relevant for here.

</TabItem>
<TabItem value="linux">

To upgrade the Svix CLI on Linux without a package manager:

1. Download the latest Linux `tar.gz` file from <https://github.com/svix/svix-cli/releases/latest>

2. Extract the file: `tar -xvf svix_X.X.X_Linux_x86_64.tar.gz`

3. Run the executable: `./svix`

</TabItem>
<TabItem value="windows">

To upgrade the Svix CLI on Windows without Scoop:

1. Download the latest Windows `tar.gz` file from <https://github.com/svix/svix-cli/releases/latest>

2. Extract the `svix_X.X.X_Windows_x86_64.tar.gz` file.

3. Run the `.exe` file

</TabItem>
</CLITabs>

### 5) Get creative

The Svix CLI has commands to interact with every part of the Svix API.

To get a complete list of commands, run `svix --help`.

Each command accepts raw JSON input as either the first positional argument or piped in over `stdin`.

With this in mind, you can chain Svix CLI commands together with other command line tools like the popular JSON parser
[jq](https://stedolan.github.io/jq/) to create powerful snippets to improve your workflow.

For example, to automatically open the dashboard for the app we created in step 3, on macOS you can run the following command:

```sh

> svix application list |  jq '.data[] | select(.name == "My Application").id' | xargs svix auth dashboard | jq '.url' | xargs open

Reviewer 2: 🔥
